haskell/src/Monocle/Backend/Test.hs

```diff
@@ -301,16 +312,38 @@ testReposSummary = withTenant doTest
             }
         ]
         results
-    query :: Q.Query
-    query =
-      let queryBH = Nothing
-          queryBHWithFlavor = const Nothing
-          queryBounds =
-            ( fromMaybe (error "nop") (readMaybe "2000-01-01 00:00:00 Z"),
-              fromMaybe (error "nop") (readMaybe "2099-12-31 23:59:59 Z")
-            )
-          queryMinBoundsSet = False
-       in Q.Query {..}
+
+testTopAuthors :: Assertion
+testTopAuthors = withTenant doTest
+  where
+    doTest :: TenantM ()
+    doTest = do
+      -- Prapare data
+      let nova = SProject "openstack/nova" [Author "alice" "a"] [Author "eve" "e"]
+      let neutron = SProject "openstack/neutron" [Author "bob" "b"] [Author "eve" "e"]
+      traverse_ (indexScenario' nova) ["42", "43"]
+      traverse_ (indexScenarioNoMerged neutron) ["142", "143"]
+
+      -- Check for expected metrics
+      results <- runQueryM defaultQuery Q.getMostActiveAuthorByChangeCreated
+      assertEqual' "Check getMostActiveAuthorByChangeCreated count" [Q.TermResult {term = "eve", count = 4}] results
+      results' <- runQueryM defaultQuery Q.getMostActiveAuthorByChangeMerged
+      assertEqual' "Check getMostActiveAuthorByChangeMerged count" [Q.TermResult {term = "eve", count = 2}] results'
+      results'' <- runQueryM defaultQuery Q.getMostActiveAuthorByChangeReviewed
+      assertEqual'
+        "Check getMostActiveAuthorByChangeReviewed count"
+        [ Q.TermResult {term = "alice", count = 2},
+          Q.TermResult {term = "bob", count = 2}
+        ]
+        results''
```

Comment:
You can create longer QueryM using do notation:

```suggestion
      runQueryM defaultQuery $ do
        results <- Q.getMostActiveAuthorByChangeCreated
        assertEqual' "Check getMostActiveAuthorByChangeCreated count" [Q.TermResult {term = "eve", count = 4}] results
        results' <- Q.getMostActiveAuthorByChangeMerged
        assertEqual' "Check getMostActiveAuthorByChangeMerged count" [Q.TermResult {term = "eve", count = 2}] results'
        results'' <- Q.getMostActiveAuthorByChangeReviewed
        assertEqual'
          "Check getMostActiveAuthorByChangeReviewed count"
          [ Q.TermResult {term = "alice", count = 2},
            Q.TermResult {term = "bob", count = 2}
          ]
          results''
```

Reply:
Ok I'll do, but to be honest I'm not sure it clarify the code or at least this way to use the do notation is too new for me.

Reply:
Using such do block is relatively common, there is even an extension to make it more easy by removing the `$`: https://typeclasses.com/ghc/block-arguments